### PR TITLE
YD-135 Replaced vpnloginid by useranonymizedid

### PR DESCRIPTION
**Pull request by aristotelos**
Only retained where it belongs in AnalysisEngineService,
LDAPUserService, VPNProfileDTO etc

### CODE REVIEW

File: core/src/main/java/nu/yona/server/subscriptions/entities/UserPrivate.java [hunk 1]
```diff
@@ -180,4 +180,9 @@ public void touch()
 	{
 		touchVersion++;
 	}
+
+	public UUID getUserAnonymizedID()
```
Comment (Bert-R) [inline core/src/main/java/nu/yona/server/subscriptions/entities/UserPrivate.java]: Shouldn't this be the replacement of ``getVPNLoginID``? In other words: shouldn't ``getVPNLoginID`` be removed?

File: core/src/main/resources/messages/messages.properties [hunk 2]
```diff
@@ -47,7 +47,7 @@ error.buddy.user.cannot.be.null=User cannot be null
 error.buddy.requesting.user.cannot.be.null=Requesting user cannot be null
 error.buddy.action.unknown=Action ''{0}'' is unknown
 error.buddy.message.entity.cannot.be.null=Message entity cannot be null
-error.buddy.vpnloginid.cannot.be.null=VPN Login Id cannot be null 
+error.buddy.useranonymizedid.cannot.be.null=Anonymized user id cannot be null
```
Comment (Bert-R) [inline core/src/main/resources/messages/messages.properties]: ID should be in capitals.

**Comment by Bert-R** [comment]
That has been quite some work. Good action!
I've added two minor comments.

File: core/src/main/java/nu/yona/server/subscriptions/entities/UserPrivate.java (hunk 1 above)
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/subscriptions/entities/UserPrivate.java]: It's referenced by `User.getVPNLoginID` which is referenced by `VPNProfileDTO.createInstance` and standing side by side to `getVPNPassword`. I don't think it should be removed.

File: core/src/main/resources/messages/messages.properties (hunk 2 above)
Comment (aristotelos) [inline core/src/main/resources/messages/messages.properties]: All the messages have id in lowercase. It's more consistent now ;). Do you want me to do a global replace?

**Comment by aristotelos** [comment]
This action was quite easy, just global search and using `Alt+Shift+R` in Eclipse...

File: core/src/main/java/nu/yona/server/subscriptions/entities/UserPrivate.java (hunk 1 above)
Comment (Bert-R) [inline core/src/main/java/nu/yona/server/subscriptions/entities/UserPrivate.java]: I see. In that case, ``getVPNLoginID`` should call ``getUserAnonymizedID`` (which would be faster, as currently ``getVPNLoginID`` first loads the object and then fetches its ID.

File: core/src/main/resources/messages/messages.properties (hunk 2 above)
Comment (Bert-R) [inline core/src/main/resources/messages/messages.properties]: Yes, please. Be careful to only do it in the text, not the message ID.

Comment (aristotelos) [inline core/src/main/resources/messages/messages.properties]: Done

File: core/src/main/java/nu/yona/server/subscriptions/entities/UserPrivate.java (hunk 1 above)
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/subscriptions/entities/UserPrivate.java]: Done

**Pull request merged**